test/test_torch.py

```diff
@@ -6777,65 +6777,93 @@ def test_neg(self, device):
+
+            matrix_inverse_out = torch.empty(n, n, dtype=torch.float64, device=device)
+            torch.inverse(matrix, out=matrix_inverse_out)
+            self.assertEqual(matrix_inverse_out, matrix_inverse, atol=0, rtol=0)
+            # second call, now that matrix_inverse_out is transposed
+            torch.inverse(matrix, out=matrix_inverse_out)
+            self.assertEqual(matrix_inverse_out, matrix_inverse, atol=0, rtol=0)
+
+            # one batch
+            matrix = random_fullrank_matrix_distinct_singular_value(n, 1).to(device)
+            matrix_inverse = torch.inverse(matrix)
+            expected_inv = matrix.squeeze(0).inverse()
+            self.assertEqual(matrix_inverse, expected_inv.unsqueeze(0))
+
+            # four batches
+            matrices = random_fullrank_matrix_distinct_singular_value(n, 4).to(device)
+            expected_inv_list = []
+            for i in range(0, 4):
+                expected_inv_list.append(torch.inverse(matrices[i]))
+            expected_inv = torch.stack(expected_inv_list)
+            matrices_inverse = torch.inverse(matrices)
+            self.assertEqual(matrices_inverse, expected_inv)
+
+            # six batches (2 x 3)
+            matrices = random_fullrank_matrix_distinct_singular_value(n, 2, 3).to(device)
+            expected_inv_list = []
+            for mat in matrices.view(-1, n, n):
+                expected_inv_list.append(torch.inverse(mat))
+            expected_inv = torch.stack(expected_inv_list).view(2, 3, n, n)
+            matrices_inverse = torch.inverse(matrices)
+            self.assertEqual(matrices_inverse, expected_inv)
+
+            # large batch size
+            # large n will be tested in test_inverse_many_batches (slow test)
+            if n == 5:
+                matrices = random_fullrank_matrix_distinct_singular_value(n, 32).to(device)
+                expected_inv_list = []
+                for i in range(0, 32):
+                    expected_inv_list.append(torch.inverse(matrices[i]))
+                expected_inv = torch.stack(expected_inv_list)
+                matrices_inverse = torch.inverse(matrices)
+                self.assertEqual(matrices_inverse, expected_inv)
+
+            # correctness test
+            matrices = random_fullrank_matrix_distinct_singular_value(n, 3).to(device)
+            matrices_inverse = torch.inverse(matrices)
+            self.assertEqual(torch.matmul(matrices, matrices_inverse), identity.expand_as(matrices))
+            self.assertEqual(torch.matmul(matrices_inverse, matrices), identity.expand_as(matrices))
+
+            # torch.inverse with out and batches
+            matrices = random_fullrank_matrix_distinct_singular_value(n, 3).to(device)
+            matrices_inverse = torch.empty(3, n, n, dtype=torch.float64, device=device)
+            torch.inverse(matrices, out=matrices_inverse)
+            self.assertEqual(torch.inverse(matrices), matrices_inverse)
```

Comment:
can you also please add couple tests for discontiguous/transposed matrix?

Reply:
will do